pkg/vmimage/vmimage_test.go

```diff
@@ -0,0 +1,75 @@
+package vmimage
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/random"
+	"github.com/openshift/openshift-azure/pkg/util/tls"
+	"github.com/sirupsen/logrus/hooks/test"
+)
+
+func TestGenerateTemplate(t *testing.T) {
+	_, hook := test.NewNullLogger()
+	timestamp := time.Now().UTC().Format("200601021504")
+	image := "rhel7-3.11-" + timestamp
+	ctx := context.Background()
+	domainNameLabel, err := random.LowerCaseAlphaString(20)
+	if err != nil {
+		t.Error(err)
+	}
+	b, err := ioutil.ReadFile("../../secrets/client-key.pem")
+	if err != nil {
+		t.Error(err)
+	}
+	clientKey, err := tls.ParsePrivateKey(b)
+	if err != nil {
+		t.Error(err)
+	}
+	b, err = ioutil.ReadFile("../../secrets/client-cert.pem")
+	if err != nil {
+		t.Error(err)
+	}
+	clientCert, err := tls.ParseCert(b)
+	if err != nil {
+		t.Error(err)
+	}
+	sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Error(err)
+	}
+	authorizer, err := azureclient.NewAuthorizerFromEnvironment("")
+	if err != nil {
+		t.Error(err)
+	}
+
+	builder := Builder{
+		GitCommit:                "unknown",
```

Comment:
please leave as many things as possible at their zero value instead of providing dummy values, it makes ongoing maintenance easier.
i.e., if the test passes when you remove the line, say, `GitCommit: ""`, remove the line, etc.

Reply:
ptal